### ISSUE TITLE
StumpWM fails to run after 4d2447244

### ISSUE THREAD

**Issue by Jehops** [stumpwm/stumpwm#1019]
After upgrading from 22.05 to 9f476737d44 StumpWM failed to load.  Bisecting tells me that 4d2447244 introduced the problem.

The is the only error message shown:
```
Type HELP for debugger help, or (SB-EXT:EXIT) to exit from SBCL.

restarts (invokable by number or by possibly-abbreviated name):
  0: [QUIT-STUMPWM] Quit Stumpwm
  1: [ABORT       ] Exit from the current thread.

(SB-PCL::INITARG-ERROR #<STANDARD-CLASS STUMPWM::HEAD> (:NUMBER))
0]
```
Is there anything else that I can test?

**Comment by jl2** [comment]
Strange.   It looks like the `number` slot of the new `frame` class doesn't have an `:initarg :number`, so `copy-frame` and `copy-head` later in the file fail with that initarg-err.

But I don't know why it would have worked in the past.

I can look into it more this evening.

**Comment by tiberious726** [comment]
The initarg problem looks like it was fixed in 2bacca53908ba3253c593bc4c9af39c1db491f90 , just a day later. Whatever is causing is causing this problem for Jehops could have been introduced by the patch series as a whole tho.

on commit 2bacca53908ba3253c593bc4c9af39c1db491f90, could you set 

```
(setq *debug-level* 5)
(redirect-all-output (data-dir-file "debug-output" "txt"))
```

early in your .stumpwmrc and post anything relevant from `~/.stumpwm.d/debug-output.txt` ?

**Comment by Jehops** [comment]
Sorry for the delay.

After more experimentation, I realized that I should have eliminated two variables when testing.  First, I was building StumpWM from the FreeBSD port and second, I was not using a default `.stumpwmrc`.  Building 90cbcbd8 using quicklisp and loading a `.stumpwmrc` with just the debugging snippet from @tiberious726 above does not result in StumpWM immediately crashing.  However, if I just run `emacsclient -nc` StumpWM does crash with this debug output.

```
17:24:06  5 Outputting a message:
        ^2*Welcome to The ^BStump^b ^BW^bindow ^BM^banager!
        Press ^5*C-t ?^2* for help.
17:24:06  3 Group windows: NIL
17:24:06  3 NO-FOCUS called
17:24:06  5 show-frame-outline!
17:24:06  5 Redrawing message window!
17:24:06  5 Outputting a message:
        ^2*Welcome to The ^BStump^b ^BW^bindow ^BM^banager!
        Press ^5*C-t ?^2* for help.
17:24:06  1 >>> :CREATE-NOTIFY
17:24:06  2 <<< :CREATE-NOTIFY
17:24:06  1 >>> :CREATE-NOTIFY
17:24:06  2 <<< :CREATE-NOTIFY
17:24:06  1 >>> :CREATE-NOTIFY
17:24:06  2 <<< :CREATE-NOTIFY
17:24:06  1 >>> :CREATE-NOTIFY
17:24:06  2 <<< :CREATE-NOTIFY
17:24:06  1 >>> :CREATE-NOTIFY
17:24:06  2 <<< :CREATE-NOTIFY
17:24:06  1 >>> :MAP-NOTIFY
17:24:06  2 <<< :MAP-NOTIFY
17:24:06  1 >>> :MAP-NOTIFY
17:24:06  2 <<< :MAP-NOTIFY
17:24:06  1 >>> :PROPERTY-NOTIFY
17:24:06  2 property notify #<XLIB:WINDOW :0 13A> :_NET_SUPPORTED :NEW-VALUE
17:24:06  2 <<< :PROPERTY-NOTIFY
17:24:06  1 >>> :PROPERTY-NOTIFY
17:24:06  2 property notify #<XLIB:WINDOW :0 13A> :_NET_SUPPORTING_WM_CHECK :NEW-VALUE
17:24:06  2 <<< :PROPERTY-NOTIFY
17:24:06  1 >>> :PROPERTY-NOTIFY
17:24:06  2 property notify #<XLIB:WINDOW :0 13A> :_NET_CLIENT_LIST :NEW-VALUE
17:24:06  2 <<< :PROPERTY-NOTIFY
17:24:06  1 >>> :PROPERTY-NOTIFY
17:24:06  2 property notify #<XLIB:WINDOW :0 13A> :_NET_DESKTOP_GEOMETRY :NEW-VALUE
17:24:06  2 <<< :PROPERTY-NOTIFY
17:24:06  1 >>> :PROPERTY-NOTIFY
17:24:06  2 property notify #<XLIB:WINDOW :0 13A> :_NET_DESKTOP_VIEWPORT :NEW-VALUE
17:24:06  2 <<< :PROPERTY-NOTIFY
17:24:06  1 >>> :PROPERTY-NOTIFY
17:24:06  2 property notify #<XLIB:WINDOW :0 13A> :_NET_NUMBER_OF_DESKTOPS :NEW-VALUE
17:24:06  2 <<< :PROPERTY-NOTIFY
17:24:06  1 >>> :PROPERTY-NOTIFY
17:24:06  2 property notify #<XLIB:WINDOW :0 13A> :_NET_DESKTOP_NAMES :NEW-VALUE
17:24:06  2 <<< :PROPERTY-NOTIFY
17:24:06  1 >>> :CONFIGURE-NOTIFY
17:24:06  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000002> 0 0 1 1 1 NIL 
17:24:06  2 <<< :CONFIGURE-NOTIFY
17:24:06  1 >>> :CONFIGURE-NOTIFY
17:24:06  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000002> 0 0 334 30 1 NIL 
17:24:06  2 <<< :CONFIGURE-NOTIFY
17:24:06  1 >>> :CONFIGURE-NOTIFY
17:24:06  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000002> 1030 0 334 30 1 NIL 
17:24:06  2 <<< :CONFIGURE-NOTIFY
17:24:06  1 >>> :MAP-NOTIFY
17:24:06  2 <<< :MAP-NOTIFY
17:24:06  1 >>> :EXPOSURE
17:24:06  2 <<< :EXPOSURE
17:24:06  1 >>> :NO-EXPOSURE
17:24:06  2 <<< :NO-EXPOSURE
17:24:06  1 >>> :NO-EXPOSURE
17:24:06  2 <<< :NO-EXPOSURE
17:24:07  1 >>> :KEY-PRESS
17:24:07  5 vvv Grab keyboard: :SUCCESS
17:24:07  1 Awaiting key (*TOP-MAP* *TILE-GROUP-TOP-MAP* *GROUP-TOP-MAP*)
17:24:07  1 key-press: #S(STUMPWM::KEY
              :KEYSYM 116
              :SHIFT NIL
              :CONTROL T
              :META NIL
              :ALT NIL
              :HYPER NIL
              :SUPER NIL
              :ALTGR NIL) 4 *ROOT-MAP*
17:24:08  1 Awaiting key (*ROOT-MAP* *TILE-GROUP-ROOT-MAP* *GROUP-ROOT-MAP*)
17:24:08  1 key-press: #S(STUMPWM::KEY
              :KEYSYM 33
              :SHIFT NIL
              :CONTROL NIL
              :META NIL
              :ALT NIL
              :HYPER NIL
              :SUPER NIL
              :ALTGR NIL) 1 "exec"
17:24:08  5 ^^^ Ungrab keyboard: 315
17:24:08  5 vvv Grab keyboard: :SUCCESS
17:24:16  5 ^^^ Ungrab keyboard: 1212
17:24:16  2 <<< :KEY-PRESS
17:24:16  1 >>> :UNMAP-NOTIFY
17:24:16  2 UNMAP: NIL T NIL
17:24:16  2 <<< :UNMAP-NOTIFY
17:24:16  1 >>> :EXPOSURE
17:24:16  5 show-frame-outline!
17:24:16  2 <<< :EXPOSURE
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 0 0 209 15 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1155 0 209 15 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1155 0 209 15 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :MAP-NOTIFY
17:24:16  2 <<< :MAP-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1155 0 209 1515 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1155 0 209 75 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :EXPOSURE
17:24:16  5 show-frame-outline!
17:24:16  2 <<< :EXPOSURE
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1155 0 209 60 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :EXPOSURE
17:24:16  5 show-frame-outline!
17:24:16  2 <<< :EXPOSURE
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1155 0 209 30 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :EXPOSURE
17:24:16  5 show-frame-outline!
17:24:16  2 <<< :EXPOSURE
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1155 0 217 30 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1147 0 217 30 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1147 0 226 30 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1138 0 226 30 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1138 0 235 15 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :EXPOSURE
17:24:16  5 show-frame-outline!
17:24:16  2 <<< :EXPOSURE
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1129 0 235 15 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1129 0 244 15 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1120 0 244 15 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1120 0 253 15 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1111 0 253 15 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1111 0 262 15 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000003> 1102 0 262 15 1 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :UNMAP-NOTIFY
17:24:16  2 UNMAP: NIL T NIL
17:24:16  2 <<< :UNMAP-NOTIFY
17:24:16  1 >>> :EXPOSURE
17:24:16  5 show-frame-outline!
17:24:16  2 <<< :EXPOSURE
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 13A> :GDK_VISUALS :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :CREATE-NOTIFY
17:24:16  2 <<< :CREATE-NOTIFY
17:24:16  1 >>> :CREATE-NOTIFY
17:24:16  2 <<< :CREATE-NOTIFY
17:24:16  1 >>> :CLIENT-MESSAGE
17:24:16  2 client message: :_NET_WM_STATE #(0 402 0 1 0)
17:24:16  2 <<< :CLIENT-MESSAGE
17:24:16  1 >>> :CLIENT-MESSAGE
17:24:16  2 client message: :_NET_WM_STATE #(0 403 0 1 0)
17:24:16  2 <<< :CLIENT-MESSAGE
17:24:16  1 >>> :CONFIGURE-REQUEST
17:24:16  3 CONFIGURE REQUEST :ABOVE #<XLIB:WINDOW :0 120001C> 0 0 736 646 0 12 
17:24:16  2 <<< :CONFIGURE-REQUEST
17:24:16  1 >>> :CLIENT-MESSAGE
17:24:16  2 client message: :_NET_WM_STATE #(0 374 0 1 0)
17:24:16  2 <<< :CLIENT-MESSAGE
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 120001C> 0 0 736 646 0 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-REQUEST
17:24:16  3 CONFIGURE REQUEST :ABOVE #<XLIB:WINDOW :0 120001C> 0 0 736 646 0 64 
17:24:16  2 <<< :CONFIGURE-REQUEST
17:24:16  1 >>> :MAP-REQUEST
17:24:16  3 map request: #<WINDOW :0 120001C> #<WINDOW :0 13A> NIL
17:24:16  3 Free number: NIL
17:24:16  3 maximizing #<TILE-WINDOW "GNU Emacs at phe.ftfl.ca" #x120001C>
17:24:16  4 maximize window #<TILE-WINDOW "GNU Emacs at phe.ftfl.ca" #x120001C> x: 0 y: 0 width: 1360 height: 765 border: 0 stick: NIL
17:24:16  3 focus-window: #<TILE-WINDOW "GNU Emacs at phe.ftfl.ca" #x120001C>
17:24:16  4 Ignoring error: XLIB:MATCH-ERROR
17:24:16  5 show-frame-outline!
17:24:16  3 focus-window: #<TILE-WINDOW "GNU Emacs at phe.ftfl.ca" #x120001C>
17:24:16  5 show-frame-outline!
17:24:16  4 Ignoring error: XLIB:MATCH-ERROR
17:24:16  2 <<< :MAP-REQUEST
17:24:16  1 >>> :CONFIGURE-REQUEST
17:24:16  3 CONFIGURE REQUEST :ABOVE #<XLIB:WINDOW :0 120001C> 0 0 736 646 0 64 
17:24:16  3 focus-window: #<TILE-WINDOW "GNU Emacs at phe.ftfl.ca" #x120001C>
17:24:16  5 show-frame-outline!
17:24:16  4 Ignoring error: XLIB:MATCH-ERROR
17:24:16  2 <<< :CONFIGURE-REQUEST
17:24:16  1 >>> :MAP-REQUEST
17:24:16  3 map request: #<WINDOW :0 120001C> #<WINDOW :0 13A> #<TILE-WINDOW "GNU Emacs at phe.ftfl.ca" #x120001C>
17:24:16  1 map request for mapped window #<TILE-WINDOW "GNU Emacs at phe.ftfl.ca" #x120001C>
17:24:16  2 <<< :MAP-REQUEST
17:24:16  1 >>> :CLIENT-MESSAGE
17:24:16  2 client message: :_NET_WM_STATE #(0 374 0 1 0)
17:24:16  2 <<< :CLIENT-MESSAGE
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 13A> :_NET_CLIENT_LIST :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 13A> :_NET_CLIENT_LIST_STACKING :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :CREATE-NOTIFY
17:24:16  2 <<< :CREATE-NOTIFY
17:24:16  1 >>> :REPARENT-NOTIFY
17:24:16  2 <<< :REPARENT-NOTIFY
17:24:16  1 >>> :REPARENT-NOTIFY
17:24:16  2 <<< :REPARENT-NOTIFY
17:24:16  1 >>> :REPARENT-NOTIFY
17:24:16  2 <<< :REPARENT-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :_NET_WM_ALLOWED_ACTIONS :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :WM_STATE :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :_NET_WM_DESKTOP :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 120001C> 3 1 1360 765 0 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 120001C> 3 1 1360 765 0 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 100000C> 0 0 736 646 0 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 100000C> 0 0 1366 646 0 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :CONFIGURE-NOTIFY
17:24:16  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 100000C> 0 0 1366 768 0 NIL 
17:24:16  2 <<< :CONFIGURE-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :_NET_FRAME_EXTENTS :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :MAP-NOTIFY
17:24:16  2 <<< :MAP-NOTIFY
17:24:16  1 >>> :MAP-NOTIFY
17:24:16  2 <<< :MAP-NOTIFY
17:24:16  1 >>> :MAP-NOTIFY
17:24:16  2 <<< :MAP-NOTIFY
17:24:16  1 >>> :ENTER-NOTIFY
17:24:16  2 <<< :ENTER-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :WM_STATE :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :_NET_WM_STATE :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :FOCUS-IN
17:24:16  2 <<< :FOCUS-IN
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 13A> :_NET_ACTIVE_WINDOW :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 13A> :_NET_CLIENT_LIST_STACKING :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :CLIENT-MESSAGE
17:24:16  2 client message: :_NET_WM_STATE #(1 405 404 1 0)
17:24:16  2 <<< :CLIENT-MESSAGE
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :_NET_WM_NAME :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :WM_NAME :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :_NET_WM_ICON_NAME :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :WM_ICON_NAME :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :WM_ICON_NAME :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :_NET_WM_ICON_NAME :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :_NET_WM_ICON :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :WM_HINTS :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :FOCUS-OUT
17:24:16  5 #<XLIB:WINDOW :0 120001C> :NORMAL :INFERIOR 
17:24:16  2 <<< :FOCUS-OUT
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :_NET_WM_OPAQUE_REGION :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :CREATE-NOTIFY
17:24:16  2 <<< :CREATE-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :_NET_WM_ICON :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
17:24:16  1 >>> :PROPERTY-NOTIFY
17:24:16  2 property notify #<XLIB:WINDOW :0 120001C> :WM_HINTS :NEW-VALUE
17:24:16  2 <<< :PROPERTY-NOTIFY
```
Starting with [my usual configuration](https://github.com/Jehops/config/blob/master/stumpwm/.stumpwmrc), but commenting out the code for the modeline, StumpWM also does not immediately crash, but if I leave that snippet in it crashes with this debug out.

```
17:35:52  3 modeline: #S(MODE-LINE
             :SCREEN #<SCREEN #<XLIB:SCREEN :0.0 1366x768x24 TRUE-COLOR>>
             :HEAD #<HEAD-FRAME 0 NIL 0 0 1366 768>
             :WINDOW #<XLIB:WINDOW :0 1000009>
             :FORMAT *SCREEN-MODE-LINE-FORMAT*
             :POSITION :BOTTOM
             :CONTENTS "^[^B^7*phe^] ^[^8*C^]    0^[^9*MHz^] 0^[^9*?C^] ^[^8*M^] ^[^9*%^] ^[^8*^]   0/  0^[^9*GiB^]   0^[^9*GiB^]   0^[^9*%^]   0.0^[^9*MiB/s^]   0.0^[^9*MiB/s^] ^[^8*^] 0       ^[^9*KiB/s^] 0       ^[^9*KiB/s^] ^[^8*B^] ^[^1*  0^] ^[^8*V^] .00 ^[^B^1*NIL^] ^[^B^7*Thu Aug 11 17:35:52^] ^[^B^4*^]"
             :CC #S(CCONTEXT
                    :SCREEN #<SCREEN #<XLIB:SCREEN :0.0 1366x768x24 TRUE-COLOR>>
                    :WIN #<XLIB:WINDOW :0 1000009>
                    :PX #<XLIB:PIXMAP :0 100000B>
                    :GC #<XLIB:GCONTEXT :0 16777226>
                    :DEFAULT-FG 8355711
                    :DEFAULT-BRIGHT 12566463
                    :DEFAULT-BG 0
                    :FG NIL
                    :BG NIL
                    :BRIGHTP NIL
                    :REVERSEP NIL
                    :COLOR-STACK NIL
                    :FONT #<XLIB:FONT 9x15 :0 16777217>)
             :HEIGHT 17
             :FACTOR 751/768
             :MODE :STUMP
             :ON-CLICK-BOUNDS NIL
             :NEW-BOUNDS NIL)
17:35:52  3 Group windows: NIL
17:35:52  3 NO-FOCUS called
17:35:52  5 show-frame-outline!
17:35:52  5 Redrawing message window!
17:35:52  1 >>> :CREATE-NOTIFY
17:35:52  2 <<< :CREATE-NOTIFY
17:35:52  1 >>> :CREATE-NOTIFY
17:35:52  2 <<< :CREATE-NOTIFY
17:35:52  1 >>> :CREATE-NOTIFY
17:35:52  2 <<< :CREATE-NOTIFY
17:35:52  1 >>> :CREATE-NOTIFY
17:35:52  2 <<< :CREATE-NOTIFY
17:35:52  1 >>> :CREATE-NOTIFY
17:35:52  2 <<< :CREATE-NOTIFY
17:35:52  1 >>> :MAP-NOTIFY
17:35:52  2 <<< :MAP-NOTIFY
17:35:52  1 >>> :MAP-NOTIFY
17:35:52  2 <<< :MAP-NOTIFY
17:35:52  1 >>> :PROPERTY-NOTIFY
17:35:52  2 property notify #<XLIB:WINDOW :0 13A> :_NET_SUPPORTED :NEW-VALUE
17:35:52  2 <<< :PROPERTY-NOTIFY
17:35:52  1 >>> :PROPERTY-NOTIFY
17:35:52  2 property notify #<XLIB:WINDOW :0 13A> :_NET_SUPPORTING_WM_CHECK :NEW-VALUE
17:35:52  2 <<< :PROPERTY-NOTIFY
17:35:52  1 >>> :PROPERTY-NOTIFY
17:35:52  2 property notify #<XLIB:WINDOW :0 13A> :_NET_CLIENT_LIST :NEW-VALUE
17:35:52  2 <<< :PROPERTY-NOTIFY
17:35:52  1 >>> :PROPERTY-NOTIFY
17:35:52  2 property notify #<XLIB:WINDOW :0 13A> :_NET_DESKTOP_GEOMETRY :NEW-VALUE
17:35:52  2 <<< :PROPERTY-NOTIFY
17:35:52  1 >>> :PROPERTY-NOTIFY
17:35:52  2 property notify #<XLIB:WINDOW :0 13A> :_NET_DESKTOP_VIEWPORT :NEW-VALUE
17:35:52  2 <<< :PROPERTY-NOTIFY
17:35:52  1 >>> :PROPERTY-NOTIFY
17:35:52  2 property notify #<XLIB:WINDOW :0 13A> :_NET_NUMBER_OF_DESKTOPS :NEW-VALUE
17:35:52  2 <<< :PROPERTY-NOTIFY
17:35:52  1 >>> :PROPERTY-NOTIFY
17:35:52  2 property notify #<XLIB:WINDOW :0 13A> :_NET_DESKTOP_NAMES :NEW-VALUE
17:35:52  2 <<< :PROPERTY-NOTIFY
17:35:52  1 >>> :CREATE-NOTIFY
17:35:52  2 <<< :CREATE-NOTIFY
17:35:52  1 >>> :CONFIGURE-NOTIFY
17:35:52  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000009> 0 0 1 17 0 NIL 
17:35:52  2 <<< :CONFIGURE-NOTIFY
17:35:52  1 >>> :CONFIGURE-NOTIFY
17:35:52  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000009> 0 0 1366 17 0 NIL 
17:35:52  2 <<< :CONFIGURE-NOTIFY
17:35:52  1 >>> :CONFIGURE-NOTIFY
17:35:52  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000009> 0 751 1366 17 0 NIL 
17:35:52  2 <<< :CONFIGURE-NOTIFY
17:35:52  1 >>> :MAP-NOTIFY
17:35:52  2 <<< :MAP-NOTIFY
17:35:52  1 >>> :EXPOSURE
17:35:52  2 <<< :EXPOSURE
17:35:52  1 >>> :CONFIGURE-NOTIFY
17:35:52  4 CONFIGURE NOTIFY NIL #<XLIB:WINDOW :0 1000009> 0 751 1366 17 0 NIL 
17:35:52  2 <<< :CONFIGURE-NOTIFY
17:35:52  1 >>> :NO-EXPOSURE
17:35:52  2 <<< :NO-EXPOSURE
17:35:52  1 >>> :NO-EXPOSURE
17:35:52  2 <<< :NO-EXPOSURE
17:35:53  1 >>> :NO-EXPOSURE
17:35:53  2 <<< :NO-EXPOSURE
```

**Comment by tiberious726** [comment]
Is stump still giving you that error from your first message? I don't see any crash messages in your debug logs (they don't always make it in, but usually do for me).

The intial error message you posted mentioned heads. Are you able to connect with sly/slime? If so, could you post the output of:
`(screen-heads (current-screen))` ?

**Comment by Jehops** [comment]
As you say, the problem in 4d24472 seems to be fixed in 2bacca5.  I say this because starting stump built from 4d24472 returns me immediately to the login screen.  Running stump built on 2bacca5 seems fine (when built using quicklisp).

I am seeing other problems with multiple monitors, but those problems seem to have been introduced later than these commits.  I'm still trying to track things down and make a clearer report than this one.  I think I should close this issue and open a new one with the debugging information that you asked for.  What do you think?

**Comment by tiberious726** [comment]
If the new issues you refer to are related to changing monitors, please also test on #1010 , I've recently fixed a few of those that were affecting my setup

**Comment by Jehops** [comment]
@szos, I just reverted to sbcl-2.2.5 and built aaf1569 and, after a few minutes, stumpwm seems stable.

When I run with my usual configuration, my custom modeline modules like https://github.com/Jehops/swm-freebsd-zfs-modeline do not work though.  I'll try to determine the commit that introduces this change.

**Comment by szos** [comment]
Thats very odd; i took a quick look through the zfs modeline repo you linked and i didnt see anything that might be causing a crash or otherwise not working. When you say it doesnt work, does it crash, do the formatters not format anything, or do something else?

**Comment by Jehops** [comment]
False alarm for the modeline problems.  The formatters were not reporting valid information and they were not refreshing, but the problem was related to how I was detecting whether StumpWM was running.  Specifically, because StumpWM was now being run as an executable with SBCL embedded, [this pgrep](https://github.com/Jehops/swm-freebsd-zfs-modeline/blob/master/ml_zfs.sh#L14) was failing.  Sorry for the noise on this one.

I'm running 740e945 and I'm not seeing any problems other than a crash with xrandr which might be addressed in #1031.  The FreeBSD port is also [working again](https://cgit.freebsd.org/ports/commit/?id=8bcff3b7dc77952a94aadf0b421e421173a39546).  Thanks to everyone here for your work and help.
